Examples/DicomConvert/DicomConvert.py

```diff
@@ -0,0 +1,110 @@
+        if image.GetNumberOfComponentsPerPixel() == 1:
+            image = sitk.RescaleIntensity(image, 0, 255)
+            if image_file_reader.GetMetaData('0028|0004').strip() == 'MONOCHROME1':
+                image =  sitk.InvertIntensity(image, maximum=255)
+            image = sitk.Cast(image,sitk.sitkUInt8)
+        sitk.WriteImage(image, output_file_name)
+        return True
+    except:
+        return False
+
+
+def convert_images(input_file_names, output_file_names, new_width):
+    # Maximal number of processes in pool, which in theory we'd like to be
+    # multiprocessing.cpu_count()-1. It appears that for some systems
+    # cpu_count() does not return the correct value
+    # (https://stackoverflow.com/questions/31344582/python-multiprocessing-cpu-count-returns-1-on-4-core-nvidia-jetson-tk1).
+    # We therefore don't rely on it and set the value explicitly here.
+    MAX_PROCESSES = 15
+    with multiprocessing.Pool(processes=MAX_PROCESSES) as pool:
+        return pool.starmap(functools.partial(convert_image, new_width=new_width), zip(input_file_names, output_file_names))
+
+
+def positive_int(int_str):
+    value = int(int_str)
+    if value<=0:
+        raise argparse.ArgumentTypeError(int_str + ' is not a positive integer value')
+    return value
+
+
+def directory(dir_name):
+  if not os.path.isdir(dir_name):
+      raise argparse.ArgumentTypeError(dir_name + ' is not a valid directory name')
+  return dir_name
+
+def main(argv=None):
+
+    parser = argparse.ArgumentParser(description='Convert and resize DICOM files to common image types.')
+    parser.add_argument('root_of_data_directory', type=directory, help='Path to the topmost directory containing data.')
+    parser.add_argument('output_file_extension', help='Image file extension, this determines output file type (e.g. png) .')
+    parser.add_argument('--w', type=positive_int, help='Width of converted images.')
+    parser.add_argument('--od', type=directory, help='Output directory.')
+    args = parser.parse_args(argv)
+
+
+    input_file_names = []
+    for dir_name, subdir_names, file_names in os.walk(args.root_of_data_directory):
+        input_file_names += [os.path.join(os.path.abspath(dir_name), fname) for fname in file_names]
+    if args.od: # if all output files are written to the same directory we need them to have a unique name, so use an index.
+        file_names = [os.path.join(os.path.abspath(args.od), str(i)) for i in range(len(input_file_names))]
+    else:
+        file_names = input_file_names
+    output_file_names = [file_name+'.'+args.output_file_extension for file_name in file_names]
+
+    res = convert_images(input_file_names, output_file_names, args.w)
+    input_file_names = list(itertools.compress(input_file_names, res))
+    output_file_names = list(itertools.compress(output_file_names, res))
+
+    # save csv file mapping input and output file names
```

Comment:
It may be a few more lines of code and a loop, but you should be able to use the lower level built-in `csv` python module.

Reply:
Done.